storage-provider/server/src/p2p/register.rs

```diff
@@ -59,83 +61,100 @@ impl RegisterConfig {
             .with_swarm_config(|cfg| cfg.with_idle_connection_timeout(Duration::from_secs(10)))
             .build();
 
-        Ok((swarm, self.rendezvous_point_address, self.rendezvous_point))
+        Ok(swarm)
     }
 }
 
 /// Register the peer with the rendezvous point.
 /// The ttl is how long the peer will remain registered in seconds.
 pub(crate) async fn register(
-    mut swarm: Swarm<RegisterBehaviour>,
+    swarm: &mut Swarm<RegisterBehaviour>,
     rendezvous_point: PeerId,
     rendezvous_point_address: Multiaddr,
-    ttl: Option<u64>,
+    ttl: u64,
     namespace: Namespace,
 ) -> Result<(), P2PError> {
     tracing::info!("Attempting to register with rendezvous point {rendezvous_point} at {rendezvous_point_address}");
+    let mut register_tick = tokio::time::interval(Duration::from_secs(ttl));
+
+    // Dial into bootstrap address
     swarm.dial(rendezvous_point_address.clone())?;
+    // Get and add external address
+    let external_addr = get_external_address(swarm).await;
+    swarm.add_external_address(external_addr);
 
-    while let Some(event) = swarm.next().await {
-        match event {
-            SwarmEvent::NewListenAddr { address, .. } => {
-                tracing::info!("Listening on {}", address);
-            }
-            SwarmEvent::ConnectionClosed {
-                peer_id,
-                cause: Some(error),
-                ..
-            } if peer_id == rendezvous_point => {
-                tracing::info!("Lost connection to rendezvous point {}", error);
-            }
-            // once `/identify` did its job, we know our external address and can register
-            SwarmEvent::Behaviour(RegisterBehaviourEvent::Identify(
-                identify::Event::Received { info, .. },
-            )) => {
-                // Register our external address.
-                tracing::info!("Registering external address {}", info.observed_addr);
-                swarm.add_external_address(info.observed_addr);
-                if let Err(error) = swarm.behaviour_mut().rendezvous.register(
-                    namespace.clone(),
-                    rendezvous_point,
-                    ttl,
-                ) {
+    loop {
+        tokio::select! {
+            // Poll tick every TTL to re-register.
+            // First tick completes immediately.
+            _ = register_tick.tick() => {
+                tracing::info!("Registering with p2p node");
```

Comment:
I think @cernicc knows a bit more than me on this, but I think these logs should be contained inside an `#[instrument]` or a span to keep context